### PR TITLE
2427: add cards with special characters while specifying version

### LINKED ISSUE #2427: With Specify Version on, can't add Dandân, Séance, or other special character card names
**Describe the bug**
You can't add a card name with a special character if you've selected `Specify Version`, but you can if not:

**To Reproduce**
1. Go to a cube and hit list, then edit
2. Turn on `Specify Version`
3. Type `dandan`
4. Select any version
5. Try to add the card

**Expected behavior**
Dandan is added to the cube.

**Screenshots**
https://github.com/dekkerglen/CubeCobra/assets/12662580/e24d0c28-a471-459a-b844-9a2d41512469

**Desktop (please complete the following information):**
 - macOS
 - Chrome
 - 120.0.6099.109

### PR DESCRIPTION
**Pull request by adrianbw**
Fixes #2427 . 

Before:
https://github.com/dekkerglen/CubeCobra/assets/12662580/e24d0c28-a471-459a-b844-9a2d41512469

After:
https://github.com/dekkerglen/CubeCobra/assets/12662580/24ade1c6-010f-4f93-b592-4defe76272e4

Comments in PR. There are several changes from auto-fix.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,7 @@ jobs/export/
 public/css/bootstrap
 /temp/
 /model/
+.DS_Store
```
Comment (adrianbw) [inline .gitignore:9]: Added for macOS development.

File: serverjs/carddb.js [hunk 2]
```diff
@@ -145,29 +145,33 @@ function reasonableId(id) {
   return reasonableCard(cardFromId(id));
 }
 
+function normalizeName(name) {
+  return name
+    .trim()
+    .normalize('NFD') // convert to consistent unicode format
+    .replace(/[\u0300-\u036f]/g, '') // remove unicode
+    .toLowerCase();
+}
```
Comment (adrianbw) [inline serverjs/carddb.js:154]: Extracted this to a helper so we can use it both places.

File: serverjs/carddb.js [hunk 3]
```diff
@@ -145,29 +145,33 @@ function reasonableId(id) {
...
   }
 
-  return data.nameToId[
-    name
-      .trim()
-      .normalize('NFD') // convert to consistent unicode format
-      .replace(/[\u0300-\u036f]/g, '') // remove unicode
-      .toLowerCase()
-  ];
+  return data.nameToId[normalizeName(name)];
```
Comment (adrianbw) [inline serverjs/carddb.js:167]: Use the new helper function.

File: serverjs/carddb.js [hunk 4]
```diff
@@ -230,9 +232,7 @@ function getVersionsByOracleId(oracleId) {
   return data.oracleToId[oracleId];
 }
 
-const getReasonableCardByOracle = (oracleId) => {
-  return cardFromId(data.oracleToId[oracleId][0]);
-};
+const getReasonableCardByOracle = (oracleId) => cardFromId(data.oracleToId[oracleId][0]);
```
Comment (adrianbw) [inline serverjs/carddb.js:235]: Autofixed.

File: serverjs/carddb.js [hunk 5]
```diff
@@ -265,16 +265,12 @@ function getRelatedCards(oracleId) {
...
-          }),
-        ),
-      ];
-    }),
+    Object.entries(related).map(([label, category]) => [
+      label,
+      Object.fromEntries(
+        Object.entries(category).map(([type, indexes]) => [type, indexes.map((id) => data.indexToOracle[id])]),
+      ),
+    ]),
```
Comment (adrianbw) [inline serverjs/carddb.js:273]: Autofixed.

**Review by lunakv — CHANGES_REQUESTED** [review]
The changes look good. 👍 Just one thing, the input name should also be normalized.

With these changes, if someone wants to look for "Lórien Revealed", they can type "lorien revealed" and get a match, but if someone types the actual name because they have an "ó" symbol on their keyboard, it won't be found because you're comparing the raw (lowercased) input with a fully normalized card name.

**Comment by adrianbw** [comment]
@lunakv I fixed that in https://github.com/dekkerglen/CubeCobra/pull/2431.

**Pull request merged**